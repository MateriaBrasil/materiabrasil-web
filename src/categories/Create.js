import React from 'react'

import Checkbox from './Checkbox'
import createWithParams from './createWithParams'

export default ({ create, ...props }) => (
  <Checkbox {...props} action={createWithParams(create, props)} />
)
