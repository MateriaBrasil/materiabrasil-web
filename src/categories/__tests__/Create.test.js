import React from 'react'
import renderer from 'react-test-renderer'

import Create from '../Create'

jest.mock('croods', () => ({
  New: props => (
    <div {...props}>
      New - {props.render([{ foo: 'bar' }])} - renderCreated -{props.render([
        { foo: 'bar' },
      ])}
    </div>
  ),
}))

it('render correctly', () => {
  const props = {
    category: { name: 'foo-name' },
    parent: { id: 1234, multipleChoice: true },
  }
  const tree = renderer.create(<Create {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
