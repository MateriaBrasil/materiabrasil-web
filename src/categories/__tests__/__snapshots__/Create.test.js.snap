// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<label
  className="MuiFormControlLabel-root-1"
  style={
    Object {
      "margin": 0,
    }
  }
>
  <span
    className="MuiButtonBase-root-21 MuiIconButton-root-15 MuiSwitchBase-root-11 MuiCheckbox-root-5 MuiCheckbox-colorSecondary-10"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role={undefined}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label-20"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-24"
        color={undefined}
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <input
        autoFocus={undefined}
        checked={false}
        className="MuiSwitchBase-input-14"
        data-indeterminate={false}
        disabled={undefined}
        id={undefined}
        name="category-root-1234"
        onChange={[Function]}
        readOnly={undefined}
        required={undefined}
        tabIndex={undefined}
        type="checkbox"
        value="foo-name"
      />
    </span>
    <span
      className="MuiTouchRipple-root-68"
    />
  </span>
  <span
    className="MuiTypography-root-33 MuiTypography-body1-42 MuiFormControlLabel-label-4"
  >
    foo-name
  </span>
</label>
`;
